packages/edge-gateway/src/gateway.js

```diff
@@ -24,7 +28,17 @@ export async function gatewayGet(request, env) {
     },
   })
 
-  return getTransformedResponseWithCustomHeaders(response)
+  // Validation layer - CSP bypass
+  const resourceCid = decodeURIComponent(response.headers.get('etag') || '')
```

Comment:
could we store the "this is a good bit" on the cached response rather than having to check it in the KV every time? Like once a cid is added to the list it's very unlikely to be removed... It'd be a little more awkward, but might end up being more similar your proposal in https://github.com/nftstorage/nftstorage.link/issues/51#issuecomment-1122361502

Reply:
We don't gain much with it, or am I missing something?

On the proposal from #51 , we want to purge from cache what we don't want to serve anymore to avoid checking Denylist before serving cached content. Here, we would add temporarily to cache, but given cache is LRU it would eventually leave cache. So, we always need to check KV anyway.

If you have some angle that I am missing, can you please open an issue for us to discuss there?